ACPs/77-reinventing-subnets/README.md

```diff
@@ -46,77 +46,154 @@ For standalone networks like the Avalanche Primary Network, this is done by conn
 
-### New Registration Flow
+### Setting a Subnet Manager
 
-#### Step 1: Retrieve a BLS multisig from the Subnet
+To create a Subnet, a `CreateSubnetTx` must be issued on the P-Chain. This transaction includes an `Owner` field which defines the key that must be used to authorize any validator set additions (`AddSubnetValidatorTx`) or removals (`RemoveSubnetValidatorTx`).
 
-To provide increased optionality over Subnet Validator requirements, the P-Chain will only require an Avalanche Warp Message using the `AddressedCall` format with the below payload to join the Subnet Validator set. A `typeID` for distinguishing between the different payload types on the P-Chain must be included prior to this ACP being considered "Implementable".
+To be a Permissionless Subnet, this `Owner` key must no longer have the ability to modify the Subnet's validator set. A `ConvertSubnetTx` must first be issued to explicitly convert a Subnet from Permissioned to Permissionless. This transaction will set the `(chainID, address)` pair that will manage the Subnet going forward. After `ConvertSubnetTx` is issued, the `Owner` from the `CreateSubnetTx` that created the Subnet will no longer have the ability to modify the Subnet's validator set.
 
-```text
-+-----------+----------+-----------+
-|  subnetID : [32]byte |  32 bytes |
-+-----------+----------+-----------+
-|    nodeID : [32]byte |  32 bytes |
-+-----------+----------+-----------+
-|    weight :   uint64 |   8 bytes |
-+-----------+----------+-----------+
-|    expiry :   uint64 |   8 bytes |
-+-----------+----------+-----------+
-| signature : [64]byte |  64 bytes |
-+-----------+----------+-----------+
-                       | 144 bytes |
-                       +-----------+
+To provide maximal flexibility for Permissionless Subnets, the BLS multisignature approach in Avalanche Warp Messaging (AWM) is re-used to approve modifications to the Subnet's validator set. Using the `(chainID, address)` pair defined in the `ConvertSubnetTx`, a Warp Message with an [`AddressedCall`](https://github.com/ava-labs/avalanchego/tree/master/vms/platformvm/warp/payload#addressedcall) payload can be constructed.
+
+To validate an `AddressedCall` payload in Avalanche Warp Messaging, the `(chainID, address)` pair is used to lookup the validators of `chainID` and verify that the BLS multi-sig includes a quorum (set to 67%) of the Subnet's validator set. The P-Chain will use Warp Messages with `AddressedCall` payloads to support modifications of the Subnet's validator set using this `(chainID, address)` pair (using the `RegisterSubnetValidatorTx` and `SetSubnetValidatorWeightTx` defined later in the specification).
```

Comment:
Should it be specified here that the `sourceAddress` of the `AddressedCall` payload is expected to have 0 length?

Reply:
The `sourceAddress` is not expected to have 0 length for `RegisterSubnetValidatorTx` and `SetSubnetValidatorWeightTx`. It is expected to have 0 length for the message from P-Chain -> Subnet which is addressed in the `Proof of Subnet Validator Set Change` section